tests/testStatsdClient.cpp

```diff
@@ -110,10 +110,15 @@ void testSendRecv(uint64_t batchSize, uint64_t sendInterval) {
         throwOnError(client);
 
         // Record a gauge "titi" to 3
-        client.gauge("titi", 3);
+        client.gauge("titi", 3u);
         throwOnError(client);
         expected.emplace_back("sendRecv.titi:3|g");
 
+        // Record a gauge "titifloat" to 123.456
+        client.gauge("titifloat", -123.456f);
```

Comment:
my biggest question about this PR is about precision. single precision floats are notorious for storing essentially garbage noise after the 6th (or so) digit. converting and sending these bytes seems wasteful. maybe we should be using a fixed precision for floating point numbers? or maybe the extra noise doesnt bother anyone else?

Reply:
Added a precision config value in StatusdClient.